Vwitter/View Controllers/HomeViewController.m

```diff
@@ -19,6 +27,63 @@ @implementation HomeViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     // Do any additional setup after loading the view.
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    
+    self.refreshControl = [[UIRefreshControl alloc] init];
+    [self.refreshControl addTarget:self action:@selector(beginRefresh) forControlEvents:UIControlEventValueChanged];
+    [self.tableView insertSubview:self.refreshControl atIndex:0];
+    
+    [self loadData];
+}
+
+- (void)beginRefresh {
+    self.isRefreshing = YES;
+    [self loadData];
+}
+
+- (void)loadData {
+    PFQuery *vaQuery = [VentAudience query];
+    [vaQuery orderByDescending:@"createdAt"];
+    [vaQuery whereKey:@"userId" equalTo:[PFUser currentUser]];
+    [vaQuery includeKey:@"ventId"];
+    
+    vaQuery.limit = 20;
+    
+
+    __weak typeof(self) weakSelf = self;
+    [vaQuery findObjectsInBackgroundWithBlock:^(NSArray<VentAudience *> * _Nullable ventAudiences, NSError * _Nullable error) {
+        typeof(self) strongSelf = weakSelf;
+        if (!strongSelf) {
+            NSLog(@"I got killed!");
+            return;
+        }
+        if (ventAudiences) {
+            NSLog(@"😎😎😎 Successfully loaded home timeline");
+            // do something with the tdata fetched
+            strongSelf.arrayOfVents = [ventAudiences valueForKey:@"ventId"];
+            
+            [strongSelf.tableView reloadData];
+            
+            if (strongSelf.isRefreshing) {
+                [strongSelf.refreshControl endRefreshing];
+                strongSelf.isRefreshing = NO;
+            }
```

Comment:
This exact same refresh (end refreshing) logic exists in both the `if` and `else`, and since neither do an early return, you can easily remove it from both if/else cases and just add it below. Any time you have exactly duplicate logic, it can always be simplified.

Also - I am finishing the review for now, I will come back tomorrow to review the rest of the PR :)